### PR TITLE
Can set order value for SET_SORT action

### PR DESCRIPTION
**Pull request by frankPairs**
Hi!

This PR is related to the issue https://github.com/marmelab/react-admin/issues/2736.

Now we can pass a order value. This order value will change the state only when sort value is different than the previousState.sort.

It allows us to define a default order value every time sort value is new.

Regards!

### CODE REVIEW

File: packages/ra-core/src/reducer/admin/resource/list/queryReducer.spec.js [hunk 1]
```diff
@@ -66,4 +66,50 @@ describe('Query Reducer', () => {
...
+            const updatedState = queryReducer(
+                {},
+                {
+                    type: 'SET_SORT',
+                    payload: { sort: 'foo', order: SORT_DESC },
+                }
+            );
+            assert.deepEqual(updatedState, { sort: 'foo', order: SORT_DESC, page: 1  });
+        });
+        it('should set order with the opposite order sort value is the same llke the previous state', () => {
```
Comment (djhi) [inline packages/ra-core/src/reducer/admin/resource/list/queryReducer.spec.js]:
```suggestion
        it("should set order as the opposite of the one in previous state when sort hasn't change", () => {
```

File: packages/ra-core/src/reducer/admin/resource/list/queryReducer.spec.js [hunk 2]
```diff
@@ -66,4 +66,50 @@ describe('Query Reducer', () => {
...
+                    sort: 'foo', order: SORT_DESC, page: 1
+                },
+                {
+                    type: 'SET_SORT',
+                    payload: { sort: 'foo' },
+                }
+            );
+            assert.deepEqual(updatedState, { sort: 'foo', order: SORT_ASC, page: 1  });
+        });
+        it('should set order with the opposite order sort value is the same llke the previous state even when we pass a specific order', () => {
```
Comment (djhi) [inline packages/ra-core/src/reducer/admin/resource/list/queryReducer.spec.js]:
```suggestion
        it("should set order as the opposite of the one in previous state even if order is specified in the payload when sort hasn't change", () => {
```

**Comment by djhi** [comment]
By the way, as this is technically a new feature, would you mind targeting the `next` branch?

**Comment by frankPairs** [comment]
@djhi of course. I am going to change the target and the text titles.

thanks for reviewing

**Comment by fzaninotto** [comment]
we need to rebase `next` on `master`, as the `queryReducer` is now written in TypeScript

**Comment by djhi** [comment]
Sorry about that @frankPairs, we are migrating to TS

**Comment by frankPairs** [comment]
@djhi  @fzaninotto  it's ok, i will update the PR using TS

**Comment by frankPairs** [comment]
@djhi @fzaninotto sorry I did a rebase.

I am going to create another PR from the next branch with this changes, I think that it is simpler.

thanks

**Pull request closed without merging**